cluster/schema/meta_class.go

```diff
@@ -217,8 +217,13 @@ func (m *metaClass) AddTenants(nodeID string, req *command.AddTenantsRequest, re
 	for i, tenant := range req.Tenants {
 		names[i] = tenant.Name
 	}
+
+	if replFactor > int64(len(req.ClusterNodes)) {
+		return nil, fmt.Errorf("not enough replicas: found %d want %d", len(req.ClusterNodes), replFactor)
+	}
+
 	// First determine the partition based on the node *present at the time of the log entry being created*
-	partitions, err := m.Sharding.GetPartitions(req.ClusterNodes, names, replFactor)
+	partitions, err := m.Sharding.GetPartitions(req.ClusterNodes[:replFactor], names, replFactor)
```

Comment:
Could you leave a comment explaining why we are subslicing the cluster nodes here, as it is not clear at all why this is done.